packages/icon/src/vaadin-icon.js

```diff
@@ -325,8 +352,11 @@ class Icon extends ThemableMixin(ElementMixin(ControllerMixin(SlotStylesMixin(Ic
     // https://github.com/vaadin/web-components/issues/6301
     this.icon = '';
 
-    if (src.includes('#')) {
-      this.svg = svg`<use href="${src}"/>`;
+    if (src.includes('.svg') && (symbol || src.includes('#'))) {
```

Comment:
nit: I wonder if checking if the URL includes the file suffix is safe enough to determine if the URL is not a data URL. Might be safer to check if it starts with `data:` instead?

Reply:
Yeah, I wasn't really happy with the check. The one you suggested seems more reliable. Changed.